fix: sequential poromechanics for single-phase

paveltomin opened the pull request:
fix a bug introduced in https://github.com/GEOS-DEV/GEOS/pull/3485
basically mass that is used in accumulation term was not updated with porosity change after mechanics leading to always converged sequential outer loop

Commit:
fix: sequential poromechanics for single-phase

castelletto1 reviewed (approved):
Looks good to me, thank you!

jhuang2601 reviewed (approved):
Thanks for the fix and it works for the Mandel_Seq case.

rrsettgast committed:
Merge branch 'develop' into pt/1ph-poromech-seq-fix

paveltomin committed:
Merge branch 'develop' into pt/1ph-poromech-seq-fix

dkachuma committed:
Merge branch 'develop' into pt/1ph-poromech-seq-fix

rrsettgast committed:
Merge branch 'develop' into pt/1ph-poromech-seq-fix

dkachuma committed:
Merge branch 'develop' into pt/1ph-poromech-seq-fix

paveltomin committed:
Update baseline path in integrated tests configuration

paveltomin committed:
Update BASELINE_NOTES.md with recent PR entries

Added entries for pull requests #3880, #3299, #3279, and #3851 with details on changes and links to integrated tests.
